lib/src/client_credentials_grant.dart

```diff
@@ -40,12 +40,12 @@ import 'utils.dart';
 /// its body as a UTF-8-decoded string. It should return a map in the same
 /// format as the [standard JSON response](https://tools.ietf.org/html/rfc6749#section-5.1)
 Future<Client> clientCredentialsGrant(
-    Uri authorizationEndpoint, String identifier, String secret,
-    {Iterable<String> scopes,
+    Uri authorizationEndpoint, String? identifier, String secret,
+    {Iterable<String>? scopes,
```

Comment:
IMO this shouldn't be nullable but have `const []` as a default

Reply:
I believe it was written this way because it is optional per the RFC: https://tools.ietf.org/html/rfc6749#section-4.4.1